File: Ejercicios_POO3.1_Java/src/Ejercicio4_Estudiantes/Estudiante.java
package Ejercicio4_Estudiantes;

import java.util.ArrayList;
import java.util.List;

public class Estudiante {
    private String nombre;
    private List<Curso> cursos;

    public Estudiante(String nombre) {
        this.nombre = nombre;
        this.cursos = new ArrayList<>();
    }

    public String getNombre() {
        return nombre;
    }

    public void agregarCurso(Curso curso) {
        cursos.add(curso);
    }

    public double calcularPromedio() {
        double suma = 0;
        for (Curso curso : cursos) {
            suma += curso.getCalificacion();
        }
        return cursos.isEmpty() ? 0 : suma / cursos.size();
    }

    @Override
    public String toString() {
        return "Estudiante{" +
                "nombre='" + nombre + '\'' +
                ", promedio=" + calcularPromedio() +
                '}';
    }
}

File: Ejercicios_POO3.1_Kotlin/src/Ejercicio4_Estudiantes/Estudiante.kt
package Ejercicio4_Estudiantes


class Estudiante(val nombre: String) {
    private val cursos = mutableListOf<Curso>()

    fun agregarCurso(curso: Curso) {
        cursos.add(curso)
    }

    fun calcularPromedio(): Double {
        // Si no hay cursos, el promedio es 0 (para evitar la división por 0)
        // Si hay cursos, se calcula la suma de las calificaciones y se divide por la cantidad de cursos
        // Se utiliza el método sumOf para sumar las calificaciones de los cursos
        // Se utiliza el método size para obtener la cantidad de cursos
        // Se utiliza el método calificacion para obtener la calificación de cada curso

        return if (cursos.isEmpty()) 0.0 else cursos.sumOf { it.calificacion } / cursos.size
    }

    override fun toString(): String {
        return "Estudiante(nombre='$nombre', promedio=${calcularPromedio()})"
    }
}